api/router.go

```diff
@@ -76,7 +78,74 @@ func Serve() {
 	}
 }
 
-// GetMuxVars wraps github.com/gorilla/mux.Vars in order to mitigate context key issues in multi-repo projects.
-func GetMuxVars(r *http.Request) map[string]string {
-	return mux.Vars(r)
+type mainHandler struct {
+	mux *mux.Router
+}
+
+func (mh *mainHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	_ = module.RunWorker("http request", func(_ context.Context) error {
+		return mh.handle(w, r)
+	})
+}
+
+func (mh *mainHandler) handle(w http.ResponseWriter, r *http.Request) error {
+	// Setup context trace logging.
+	ctx, tracer := log.AddTracer(r.Context())
+	lrw := NewLoggingResponseWriter(w, r)
+	// Add request context.
+	apiRequest := &Request{
+		Request: r,
+	}
+	ctx = context.WithValue(ctx, requestContextKey, apiRequest)
+	// Add context back to request.
+	r = r.WithContext(ctx)
+
+	tracer.Tracef("api request: %s ___ %s", r.RemoteAddr, r.RequestURI)
+	defer func() {
+		// Log request status.
+		if lrw.Status != 0 {
+			// If lrw.Status is 0, the request may have been hijacked.
+			tracer.Debugf("api request: %s %d %s", lrw.Request.RemoteAddr, lrw.Status, lrw.Request.RequestURI)
+		}
+		tracer.Submit()
+	}()
+
+	// Get handler for request.
+	// Gorilla does not support handling this on our own very well.
+	// See github.com/gorilla/mux.ServeHTTP for reference.
+	var match mux.RouteMatch
+	var handler http.Handler
+	if mh.mux.Match(r, &match) {
+		handler = match.Handler
+		apiRequest.Route = match.Route
+		apiRequest.URLVars = match.Vars
+	}
+
+	// Be sure that URLVars always is a map.
+	if apiRequest.URLVars == nil {
+		apiRequest.URLVars = make(map[string]string)
+	}
+
+	// Check authentication.
+	token := authenticateRequest(lrw, r, handler)
+	if token == nil {
+		// Authenticator already replied.
+		return nil
+	}
+	apiRequest.AuthToken = &AuthToken{
+		Read:  token.Read,
+		Write: token.Write,
+	}
```

Comment:
What about:

```suggestion
	apiRequest.AuthToken = autenticateRequest(lrw, r, handler)
	if apiRequest.AuthToken == nil {
		return
	}
```
Or is a copy of the token without any expiration time the wanted effect?